pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -111,18 +113,93 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 // Name returns Name for the precondition.
 func (pf *Upgradeable) Name() string { return "ClusterVersionUpgradeable" }
 
+//  RecentEtcdBackup checks if a recent etcd backup has been taken.
+type RecentEtcdBackup struct {
+	key      string
+	lister   configv1listers.ClusterVersionLister
+	coLister configv1listers.ClusterOperatorLister
+}
+
+// NewRecentEtcdBackup returns a new RecentEtcdBackup precondition check.
+func NewRecentEtcdBackup(lister configv1listers.ClusterVersionLister, coLister configv1listers.ClusterOperatorLister) *RecentEtcdBackup {
+	return &RecentEtcdBackup{
+		key:      "version",
+		lister:   lister,
+		coLister: coLister,
+	}
+}
+
+func recentEtcdBackupCondition(lister configv1listers.ClusterOperatorLister) (string, string) {
+	var reason, message string
+	ops, err := lister.Get("etcd")
+	if err == nil {
+		backupCondition := resourcemerge.FindOperatorStatusCondition(ops.Status.Conditions, "RecentBackup")
+		if backupCondition == nil {
+			reason = "EtcdRecentBackupNotSet"
+			message = "RecentBackup: Etcd backup condition is not set."
+		} else if backupCondition.Status != configv1.ConditionTrue {
+			reason = backupCondition.Reason
+			message = backupCondition.Message
```

Comment:
The message needs to continue to contain `RecentBackup`, so etcd knows we still want a backup.  Can we `message = fmt.Sprintf("RecentBackup: %s", backupCondition.Message)` or `fmt.Sprintf("%s=%s: %s", backupCondition.Type, backupCondition.Status, backupCondition.Message)` or some such here?

Reply:
done